### DISCUSSION TITLE
Usage of kernel function for a special case

### DISCUSSION THREAD

**Discussion by niladridas** [cornellius-gp/gpytorch#2386] [General]
Hello! I need input on the following issue. 

I need to calculate the individual matrices: K1 = k(z1,z1) , K2 = k(z1,z2), K3 = k(z2,z2). 
Once I have them, my mean and covar is calculated as:
mu = K2^T(K1+epsilon*I)^{-1}y
Cov = K3 - K2^T(K1+epsilon*I)^{-1}K2 + epsilon*I

I am having difficulty in implementing it.
The length scale parameter is going to be optimized.

Thanks.

**Comment by gpleiss** [comment]
It sounds like you want to compute a GP posterior? Why not simply use an exact GP, train/condition on z1, and test on z2?
